detector/cve_client.go

```diff
@@ -155,9 +155,27 @@ func (api goCveDictClient) fetchCveDetailsByCpeName(cpeName string) ([]cvemodels
 
 		query := map[string]string{"name": cpeName}
 		logging.Log.Debugf("HTTP Request to %s, query: %#v", url, query)
-		return api.httpPost(cpeName, url, query)
+		if cves, err = api.httpPost(cpeName, url, query); err != nil {
+			return nil, err
+		}
+	} else {
+		if cves, err = api.driver.GetByCpeURI(cpeName); err != nil {
+			return nil, err
+		}
+	}
+
+	if useJVN {
+		return cves, nil
+	}
+
+	nvdCves := []cvemodels.CveDetail{}
+	for _, cve := range cves {
+		if cve.HasJvn() {
+			continue
+		}
```

Comment:
It does not seem to consider the case where both NVD and JVN are filled. (ex. NVD: CpeVersionMatch, JVN: CpeVendorProductMatch

```go
for _, cve := range cves {
    if !cve.HasNvd() {
        continue
    }
    cve.Jvn = []cvemodels.Jvn{}
    nvdCves = append(nvdCves, cve)
}
```

Reply:
It looks like they want to set an option (useJVN) to perform CPE Match in strict mode (CpeVersionMatch) or loose mode (CpeVendorProductMatch).

To do this, it is necessary to be able to classify whether the cve.Nvd is a CpeVersionMatch or a CpeVendorProductMatch.
First, I think CpeName needs to be checked. (Checking the version column)
Next, I think I need to change the response of go-cve-dictionary when the version column of CpeName is filled or not. Currently, Nvd also seems to always do a vendor product match, but I think it would be better to separate this into only a version match if the version is filled, and a vendor product match if it is not.